exporters/otlp/common/src/main/java/io/opentelemetry/exporter/otlp/internal/okhttp/OkHttpRetrier.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.exporter.otlp.internal.okhttp;
+
+import io.opentelemetry.exporter.otlp.internal.RetryPolicy;
+import java.io.IOException;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import okhttp3.Interceptor;
+import okhttp3.Response;
+
+/** Retrier of OkHttp requests. */
+public final class OkHttpRetrier implements Interceptor {
```

Comment:
`RetryInterceptor`? `OkHttp` is implied because it implements the okhttp `Interceptor` interface.